syntax = "proto3";

package multidriver;

import "google/protobuf/struct.proto";

message Empty {}

message Row {
  string id = 1;
  google.protobuf.Struct data = 2;
}

message CollectionRequest {
  string collection = 1;
}

message RowRequest {
  string collection = 1;
  string id = 2;
}

message RowID {
  string id = 1;
}

message ServiceInfo {

}

service Collection {

  rpc GetServiceInfo(CollectionRequest) returns (ServiceInfo);

  rpc GetIDs(CollectionRequest) returns (stream RowID);

  rpc GetRows(CollectionRequest) returns (stream Row);

  rpc GetRowsByID(stream RowRequest) returns (stream Row);

}
